packages/lab/src/radio-button/RadioButton.tsx

```diff
@@ -1,86 +1,156 @@
 import { clsx } from "clsx";
 import {
   ChangeEventHandler,
+  FocusEventHandler,
   forwardRef,
   HTMLAttributes,
-  ComponentType,
+  InputHTMLAttributes,
+  ReactNode,
 } from "react";
-import { makePrefixer } from "@salt-ds/core";
-import { ControlLabel, ControlLabelProps } from "../control-label";
-import { RadioButtonBase as Radio } from "./RadioButtonBase";
-import { RadioButtonIconProps } from "./RadioButtonIcon";
+import { makePrefixer, useControlled, useId } from "@salt-ds/core";
+import { useRadioGroup } from "./internal/useRadioGroup";
+import { RadioButtonIcon } from "./RadioButtonIcon";
 
 import "./RadioButton.css";
 
 const withBaseName = makePrefixer("saltRadioButton");
 
 export interface RadioButtonProps
-  extends Omit<HTMLAttributes<HTMLDivElement>, "onChange"> {
+  extends Omit<
+    HTMLAttributes<HTMLLabelElement>,
+    "onChange" | "onBlur" | "onFocus"
+  > {
+  /**
+   * Set the default selected radio button in the group
+   */
   checked?: boolean;
   className?: string;
+  /**
+   * Prop used to set the styling
+   */
   disabled?: boolean;
   /**
-   * Custom icon component
+   * Prop used to set the styling
+   */
+  error?: boolean;
```

Comment:
Is it worth making this a union? Will we need more statuses in the future?

Reply:
We didn't spec for more statuses. But that's a good point. I'll pass the idea to design and see what they think about it.

Reply:
Raised the idea with design. Depends a lot on the FormField design. I'll change this as needed if design adds some changes